src/global.cpp

```diff
@@ -36,27 +36,82 @@ int symPACK_Rank(int * rank){
 }
 
 
+#ifdef CUDA_MODE
+extern "C"
+void compute_device_alloc_size() {
+  int n_gpus = upcxx::gpu_default_device::device_n();
+  if (n_gpus==0) {
+    std::cerr<<"Error: found no CUDA devices"<<std::endl;
+    abort();
+  }
+
+  int tasks_per_node = upcxx::local_team().rank_n();
+  
+  size_t alloc_size, free, total;
+  CUDA_ERROR_CHECK(cudaMemGetInfo(&free, &total));
+  alloc_size = (free/2) / (std::max(tasks_per_node, n_gpus) / n_gpus);
```

Comment:
On a related note, why does the expression include `free/2`?
I agree with the idea that we should not attempt to allocate every last byte of available GPU memory as reported by `cudaMemGetInfo`, but discarding half the available space seems excessive. Is the balance of this GPU memory being reserved for some purpose outside the UPC++ device segment, or can we increase this to a larger fraction, eg:

```suggestion
  alloc_size = (size_t)(free*0.9) / (std::max(tasks_per_node, n_gpus) / n_gpus);
```